### PR TITLE
Add Cast update warning

### PR DESCRIPTION
**Pull request by meyskens**
Add Cast update warning on the service info page.

---
This change is [Reviewable](https://reviewable.io/reviews/innovate-technologies/control/56)

### CODE REVIEW

File: src/manage/manage-ctrl.js [hunk 1]
```diff
@@ -1,4 +1,18 @@
-export default /*@ngInject*/ function ($rootScope, service, $scope, ManageService) {
-  $rootScope.service = service;
+export default /*@ngInject*/ function ($rootScope, service, $scope, ManageService, AboutService, ConfigService) {
+  $scope.service = service;
```
Comment (leoetlino) [inline src/manage/manage-ctrl.js]: This was changed from $rootScope to $scope. I'm not sure if there's any side effect from doing this, so it's probably best to leave it.

File: src/manage/manage-ctrl.js [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+    ConfigService.getConfig().then((config) => {
+      this.castRevision = config.version.Cast;
+      checkCastVersion();
+    });
+  }
   $scope.sections = ManageService.getSections();
+
+  const checkCastVersion = () => {
+    AboutService.getCastBuildInfo().then(({ version: revision }) => {
+      if (revision !== this.castRevision) {
```
Comment (leoetlino) [inline src/manage/manage-ctrl.js]: `$scope.castNeedsUpdate = revision !== this.castRevision`

**Comment by meyskens** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/manage/manage-ctrl.js, line 2 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/control/56#-KPtTmOy7QgS_XZyeyOd-r1-2:-KPvqI0JmgQxbMV6e8WD:b-896fix) ([raw file](https://github.com/innovate-technologies/control/blob/ad4600f4e1050a567fe99cb13a5ad1fe4d97efe6/src/manage/manage-ctrl.js#L2)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> This was changed from $rootScope to $scope. I'm not sure if there's any side effect from doing this, so it's probably best to leave it.
</details>
Done.

---

*[src/manage/manage-ctrl.js, line 13 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/control/56#-KPtTmOy7QgS_XZyeyOd-r1-13:-KPvqIdW_CqjxfZzl4Kc:b-896fix) ([raw file](https://github.com/innovate-technologies/control/blob/ad4600f4e1050a567fe99cb13a5ad1fe4d97efe6/src/manage/manage-ctrl.js#L13)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> `$scope.castNeedsUpdate = revision !== this.castRevision`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/control/56)*

**Comment by leoetlino** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/manage/information-pane.html, line 2 \[r2\]](https://reviewable.io:443/reviews/innovate-technologies/control/56#-KPvqebEpdH5uA-5J0f1:-KPvqebEpdH5uA-5J0f2:bhhkgb7) ([raw file](https://github.com/innovate-technologies/control/blob/0b5169a8fa547b190136d411aaab36247ae444e4/src/manage/information-pane.html#L2)):*
> ```HTML
> <div class="alert alert-info" ng-show="service.group.toLowerCase().indexOf('node') !== -1 && castNeedsUpdate">
>     <p><fa name="info-circle"></fa> An new version of Cast is available. You can upgrade it under About.</p>
> ```

`A new version`

and

`You can update to it` or `You can update Cast`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/control/56)*

**Comment by leoetlino** [comment]
:lgtm:, just rebase (as usual) and this is good for merging.

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/control/56#-:-KPvzAsvn-K29yQuG997:be5oq34)*

**Pull request merged**